client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -81,19 +81,22 @@ class QueryComponent extends Component {
   }
 
   onBlur = evt => {
-    this.setState({ selectedQueryHold: evt.target.value, selectedQuery: evt.target.value.replaceAll("\n", " ") });
-    this.handleEditorChange(evt.target.value.replaceAll("\n", " "));
+    const selectedQuery = evt.target.value.replaceAll("\n", " ");
+    this.setState({ selectedQueryWithNewlines: evt.target.value, selectedQuery });
+    this.handleEditorChange(selectedQuery);
   }
 
   onFocus = () => {
-    const { selectedQueryHold } = this.state;
-    this.setState({ selectedQuery: selectedQueryHold });
-    this.handleEditorChange(selectedQueryHold);
+    const { selectedQueryWithNewlines } = this.state;
+    this.setState({ selectedQuery: selectedQueryWithNewlines });
+    this.handleEditorChange(selectedQueryWithNewlines);
   }
 
   handleEditorChange = value => {
-    const count = Math.min(20, Math.max(0, value.split("").filter(c => c === "\n").length));
-    this.setState({ rowHeight: `${((count + 1) * 19) + 10}px` });
+    const maxNumberOfRows = 20;
+    const lineHeight = 19;
+    const count = Math.min(maxNumberOfRows, Math.max(0, value.split("").filter(c => c === "\n").length));
+    this.setState({ rowHeight: `${((count + 1) * lineHeight) + 10}px` });
```

Comment:
Can we define what 10 is as well? Looks like some kind of min height maybe?

Reply:
Done